lib/blocs/file_download/personal_file_download_cubit.dart

```
@@ -32,23 +32,23 @@ class ProfileFileDownloadCubit extends FileDownloadCubit {
 
       emit(FileDownloadInProgress(
           fileName: file.name, totalByteCount: file.size));
-      final dataRes = await http
-          .get(_arweave.client.api.gatewayUrl.origin + '/${file.dataTxId}');
-
-      Uint8List dataBytes;
+      final dataRes = await http.get(Uri.parse(
+          _arweave.client.api.gatewayUrl.origin + '/${file.dataTxId}'));
+      late Uint8List dataBytes;
 
       if (drive.isPublic) {
-        dataBytes = await dataRes.bodyBytes;
+        dataBytes = dataRes.bodyBytes;
       } else if (drive.isPrivate) {
         final profile = _profileCubit.state as ProfileLoggedIn;
 
-        final dataTx = await _arweave.getTransactionDetails(file.dataTxId);
+        final dataTx = await (_arweave.getTransactionDetails(file.dataTxId));
 
         final fileKey =
             await _driveDao.getFileKey(driveId, fileId, profile.cipherKey);
-
-        dataBytes = await decryptTransactionData(
-            dataTx, await dataRes.bodyBytes, fileKey);
+        if (dataTx != null) {
```

Could this hide an error?

---

If there's no dataTx then that means there's no data. Not sure if that's an error.